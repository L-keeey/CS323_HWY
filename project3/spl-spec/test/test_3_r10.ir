FUNCTION main :
READ t2
&v1 := t2
t1 := &v1
t3 := &v1
t4 := #0
IF t3 > t4 GOTO label1
GOTO label2
LABEL label1 :
t5 := #1
WRITE t5
GOTO label3
LABEL label2 :
t6 := &v1
t7 := #0
IF t6 < t7 GOTO label4
GOTO label5
LABEL label4 :
t9 := #1
t8 := #0 - t9
WRITE t8
GOTO label6
LABEL label5 :
t10 := #0
WRITE t10
LABEL label6 :
LABEL label3 :
t11 := #0
RETURN t11
