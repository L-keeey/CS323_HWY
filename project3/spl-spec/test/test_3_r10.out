Program (1)
  ExtDefList (1)
    ExtDef (1)
      Specifier (1)
        TYPE: int
      FunDec (1)
        ID: main
        LP
        RP
      CompSt (2)
        LC
        DefList (3)
          Def (3)
            Specifier (3)
              TYPE: int
            DecList (3)
              Dec (3)
                VarDec (3)
                  ID: n
            SEMI
        StmtList (4)
          Stmt (4)
            Exp (4)
              Exp (4)
                ID: n
              ASSIGN
              Exp (4)
                ID: read
                LP
                RP
            SEMI
          StmtList (5)
            Stmt (5)
              IF
              LP
              Exp (5)
                Exp (5)
                  ID: n
                GT
                Exp (5)
                  INT: 0
              RP
              Stmt (5)
                Exp (5)
                  ID: write
                  LP
                  Args (5)
                    Exp (5)
                      INT: 1
                  RP
                SEMI
              ELSE
              Stmt (6)
                IF
                LP
                Exp (6)
                  Exp (6)
                    ID: n
                  LT
                  Exp (6)
                    INT: 0
                RP
                Stmt (6)
                  Exp (6)
                    ID: write
                    LP
                    Args (6)
                      Exp (6)
                        MINUS
                        Exp (6)
                          INT: 1
                    RP
                  SEMI
                ELSE
                Stmt (7)
                  Exp (7)
                    ID: write
                    LP
                    Args (7)
                      Exp (7)
                        INT: 0
                    RP
                  SEMI
            StmtList (8)
              Stmt (8)
                RETURN
                Exp (8)
                  INT: 0
                SEMI
        RC
Program
ExtDefList
ExtDef
FunDec
FUNCTION main :
ComSt
DefList
Def
DecList
Dec
VarDec
DefList
StmtList
Stmt
*******
Exp
ASSIGN
Exp
*******
Exp
assign
v1
Exp
read
READ t2
v1 := t2
t1 := v1
StmtList
Stmt
cond_Exp
3
Exp
t3 := v1
Exp
const
INT
0
t4 := #0
IF t3 > t4 GOTO label1
GOTO label2
LABEL label1 :
Stmt
*******
ID
LP
Args
RP
*******
Exp
Exp
const
INT
0
t5 := #1
WRITE t5
GOTO label3
LABEL label2 :
Stmt
cond_Exp
3
Exp
t6 := v1
Exp
const
INT
0
t7 := #0
IF t6 < t7 GOTO label4
GOTO label5
LABEL label4 :
Stmt
*******
ID
LP
Args
RP
*******
Exp
Exp
Exp
const
INT
0
t9 := #1
t8 := #0 - t9
WRITE t8
GOTO label6
LABEL label5 :
Stmt
*******
ID
LP
Args
RP
*******
Exp
Exp
const
INT
0
t10 := #0
WRITE t10
LABEL label6 :
LABEL label3 :
StmtList
Stmt
Exp
const
INT
0
t11 := #0
RETURN t11
StmtList
ExtDefList
FUNCTION main :
READ t2
v1 := t2
t1 := v1
t3 := v1
IF t3 > t4 GOTO label1
GOTO label2
LABEL label1 :
WRITE t5
GOTO label3
LABEL label2 :
t6 := v1
t7 := #0
IF t6 < t7 GOTO label4
GOTO label5
LABEL label4 :
t9 := #1
t8 := #0 - t9
WRITE t8
GOTO label6
LABEL label5 :
t10 := #0
WRITE t10
LABEL label6 :
LABEL label3 :
t11 := #0
RETURN t11
